controlplane/src/core/repositories/analytics/UsageRepository.ts

```diff
@@ -161,4 +161,133 @@ export class UsageRepository {
       meta,
     };
   }
+
+  public async getUnusedFields({
+    organizationId,
+    federatedGraphId,
+    range,
+    fields,
+  }: {
+    organizationId: string;
+    federatedGraphId: string;
+    range: number;
+    fields: Field[];
+  }): Promise<{ name: string; typeName: string }[]> {
+    const arrayJoinFields = fields.map((field) => `('${field.name}', '${field.typeName}')`).join(', ');
+    const {
+      dateRange: { end, start },
+    } = parseTimeFilters(undefined, range);
+
+    const query = `
+      WITH 
+        toDateTime('${start}') AS startDate,
+        toDateTime('${end}') AS endDate,
+        all_fields AS (
+          SELECT
+              field.1 as Name,
+              field.2 as TypeName
+          FROM
+          (
+            SELECT
```

Comment:
We don't filter the dataset by date, org, graph here. This can be very expensive.

Reply:
We do filter by time, org and fed graph

Reply:
But not in those sub queries. You only filter in the most outer query.